File: maxRange/dp.py
# debugの時はファイルから読み込む方が楽
# 環境によっては、 `maxRange/input/input1.txt` などに変更しましょう
#import sys; sys.stdin = open('input/input1.txt', 'r')


def main():
    n = int(input())
    numbers = [int(input()) for _ in range(n)]
    max_s = 0
    si = 0
    for li in numbers:
        prev_si = si
        s = prev_si + li
        si = s if s > 0 else 0
        max_s = max(max_s, si)
    print(max_s)


main()

File: maxRange/Dp.java
package maxRange;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Scanner;

class Dp {
    public static void main(String[] args) throws IOException {
        // debug 時はファイルから読み込みのが楽
        // System.setIn(new FileInputStream("maxRange/input/input1.txt"));
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int[] numbers = new int[n];
        for(int i=0; i<n; i++){
            numbers[i] = input.nextInt();
        }
        int maxS = 0;
        int Si = 0;

        for (int i = 0; i < n; i++) {
            int prevSi = Si;
            int s = prevSi + numbers[i];
            if (s > 0)
                Si = s;
            else
                Si = 0;
            if (Si > maxS) maxS = Si;
        }
        System.out.println(maxS);
    }
}